docs/build-with-0g/compute-network/sdk.md

```diff
@@ -7,179 +7,142 @@ sidebar_position: 3
+```typescript
+import { BrowserProvider } from "ethers";
 import { createZGServingNetworkBroker } from "@0glabs/0g-serving-broker";
 
-/**
- * 'createZGServingNetworkBroker' is used to initialize ZGServingUserBroker
- *
- * @param {JsonRpcSigner | Wallet} signer - A signer that implements the 'JsonRpcSigner' or
- * 'Wallet' interface from the ethers package.
- *
- * @param {string} contractAddress - 0G Serving contract address, use default address if not provided.
- *
- * @returns broker instance.
- *
- * @throws An error if the broker cannot be initialized.
- */
-const broker = await createZGServingNetworkBroker(signer);
+async function initializeBrokerWithWallet() {
+  // Ensure wallet is installed
+  if (typeof window.ethereum !== 'undefined') {
+    // Create a provider
+    const provider = new BrowserProvider(window.ethereum);
+    
+    // Get the signer
+    const signer = await provider.getSigner();
+    
+    // Initialize broker
+    const broker = await createZGServingNetworkBroker(signer);
+  }
+}
 ```
 
-### Step 3: List Available Services
+> 💡 The `contractAddress` parameter is optional - the SDK uses a default address if not provided.
+
+### Discover Available Services
+
+The 0G Compute Network hosts multiple AI service providers. The service discovery process helps you find and select the appropriate services for your needs.
 
 ```typescript
-/**
- * 'listService' retrieves a list of services from the contract.
- *
- * @returns {Promise<ServiceStructOutput[]>} A promise that resolves to an array of ServiceStructOutput
- * objects.
- *
- * type ServiceStructOutput = {
- *   provider: string;  // Provider's wallet address, which is the unique identifier for the provider.
- *   name: string;
- *   serviceType: string;
- *   url: string;
- *   inputPrice: bigint;
- *   outputPrice: bigint;
- *   updatedAt: bigint;
- *   model: string;
- * };
- *
- * @throws An error if the service list cannot be retrieved.
- */
 const services = await broker.listService();
 ```
 
-### Step 4: Manage Accounts
+Each service contains the following information:
+```typescript
+type ServiceStructOutput = {
+  provider: string;  // Provider's wallet address (unique identifier)
+  name: string;      // Service name
+  serviceType: string; // Type of service
+  url: string;       // Service URL
+  inputPrice: bigint; // Price for input processing
+  outputPrice: bigint; // Price for output generation
+  updatedAt: bigint;  // Last update timestamp
+  model: string;      // Model identifier
+};
+```
 
-Before using the provider's services, you need to create an account specifically for the chosen provider. The provider checks the account balance before responding to requests. If the balance is insufficient, the request will be denied.
+### Account Management
 
-#### 4.1 Create an Account
+The 0G Compute Network uses a prepaid account system for each provider. Before using any services, you need to set up and fund an account for each provider you want to use.
```

Comment:
Does it make sense to swap #4 create an account with #3 list available services since you need to create an account first?

Reply:
We need the provider address, to create the account on the specific provider so, It will be better to list the service first to get the provider address